pkgs/os-specific/darwin/coconutbattery/default.nix

```
@@ -0,0 +1,41 @@
+{ lib
+, stdenvNoCC
+, fetchzip
+}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "coconutbattery";
+  version = "3.9.12";
+
+  src = fetchzip {
+    url = "https://coconut-flavour.com/downloads/coconutBattery_${lib.strings.stringAsChars (x: if x == "." then "" else x) version}.zip";
```

```suggestion
    url = "https://coconut-flavour.com/downloads/coconutBattery_${builtins.replaceStrings [ "." ] [ "" ] version}.zip";
```
I think `builtins.replaceStrings` is often used for such replacements.